java-frontend/src/main/java/org/sonar/java/model/DefaultJavaFileScannerContext.java

```diff
@@ -109,29 +112,39 @@ public void addIssue(File file, JavaCheck check, int line, String message) {
 
   @Override
   public void reportIssue(JavaCheck javaCheck, Tree tree, String message) {
+    throwIfCrossFileScanner(javaCheck);
     reportIssue(javaCheck, tree, message, ImmutableList.of(), null);
   }
 
   @Override
   public void reportIssue(JavaCheck javaCheck, Tree syntaxNode, String message, List<Location> secondary, @Nullable Integer cost) {
+    throwIfCrossFileScanner(javaCheck);
```

Comment:
`reportIssue(AnalyzerMessage message)` is called twice in this class, so I have two options:

* I throw only in those two methods - line 127 and line 139 in this file
* I throw in all five public methods

I choose to throw in all, because the intent is more clear like this. Moreover, other public classes might change and call `reportIssue(AnalyzerMessage message)` directly, and would then need to do the throw-check